src/app/static/src/app/store/adrUpload/mutations.ts

```diff
@@ -34,6 +40,30 @@ export const mutations: MutationTree<ADRUploadState> = {
         state.totalFilesUploading = null;
     },
 
+    [ADRUploadMutation.ReleaseCreated](state: ADRUploadState) {
+        state.releaseCreated = true;
+    },
+
+    [ADRUploadMutation.ClearStatus](state: ADRUploadState) {
+        state.releaseCreated = false;
+        state.releaseFailed = false;
+        state.uploadComplete = false;
+    },
+
+    [ADRUploadMutation.ReleaseFailed](state: ADRUploadState, action: PayloadWithType<Error | null>) {
+        state.releaseFailed = true;
+        const alteredPayload = action.payload
+        switch (action.payload?.detail) {
+            case "Version already exists for this activity":
+                alteredPayload!.detail = i18next.t("releaseExists", this.currentLanguage)
```

Comment:
Sorry - I misled you here. The following should be sufficient:
```suggestion
                alteredPayload!.detail = i18next.t("releaseExists");
```
What I suggested only works by accident: `this.currentLanguage` is a copy-and-paste that isn't valid (it's not defined here) but works because it's therefore ignored by the `t()` function, which instead uses the currently set language - which is fine. Same for L61.

In fact, I'm not sure why we ever pass a second argument to this function, because we always call `i18next.changeLanguage()` when the language changes - which should be sufficient.